wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -20,32 +20,120 @@
+ * and enums that are (transitive) dependencies of the listed names. The {@code --excludes} flag
+ * excludes types, and takes precedence over {@code --includes}.
+ *
+ * <p>If the {@code --registry_class} flag is present, its argument must be a Java class name. A
+ * class with the given name will be generated, containing a constant list of all extension
+ * classes generated during the compile. This list is suitable for passing to Wire's constructor
+ * at runtime for constructing its internal extension registry.
+ *
+ * <p>Unless the {@code --no_options} flag is supplied, code will be emitted for options on messages
+ * and fields.  The presence of options on a message will result in a static member named
+ * "MESSAGE_OPTIONS", initialized with the options and their values.   The presence of options on
+ * a field (other than the standard options "default", "deprecated", and "packed") will result in
+ * a static member named "FIELD_OPTIONS_&lt;field name&gt;" in the generated code, initialized
+ * with the field option values.
+ *
+ * <p>If {@code --quiet} is specified, diagnostic messages to stdout are suppressed.
+ *
+ * <p>The {@code --dry_run} flag causes the compile to just emit the names of the source files that
+ * would be generated to stdout.
+ *
+ * <p>The {@code --android} flag will cause all messages to implement the {@code Parcelable}
+ * interface.
+ *
+ * <p>The {@code --compact} flag will emit code that uses reflection for reading, writing, and
+ * toString methods which are normally implemented with code generation.
+ */
 public final class WireCompiler {
+  public static final String PROTO_PATH_FLAG = "--proto_path=";
+  public static final String JAVA_OUT_FLAG = "--java_out=";
+  public static final String FILES_FLAG = "--files=";
+  public static final String ROOTS_FLAG = "--roots="; // TODO(jwilson): drop before 2.0 final.
+  public static final String INCLUDES_FLAG = "--includes=";
+  public static final String EXCLUDES_FLAG = "--excludes=";
+  public static final String NO_OPTIONS_FLAG = "--no_options";
+  public static final String QUIET_FLAG = "--quiet";
+  public static final String DRY_RUN_FLAG = "--dry_run";
+  public static final String ANDROID = "--android";
+  public static final String COMPACT = "--compact";
+
   private static final String CODE_GENERATED_BY_WIRE =
       "Code generated by Wire protocol buffer compiler, do not edit.";
 
   private final FileSystem fs;
-  private final CommandLineOptions options;
   private final WireLogger log;
 
-  /**
-   * Runs the compiler.  See {@link CommandLineOptions} for command line options.
-   */
+  final List<String> protoPaths;
+  final String javaOut;
+  final List<String> sourceFileNames;
+  final IdentifierSet identifierSet;
+  final boolean emitOptions;
+  final boolean dryRun;
+  final boolean emitAndroid;
+  final boolean emitCompact;
+
+  public WireCompiler(FileSystem fs, WireLogger log, List<String> protoPaths, String javaOut,
```

Comment:
private or package scope this guy? historically it's changed over the course of a major version when new major features come in.